django/db/models/options.py

```diff
@@ -280,6 +280,10 @@ def _prepare(self, model):
                 if already_created:
                     field = already_created[0]
                 field.primary_key = True
+                try:
+                    field.auto_created = field.reverse_related_fields[0][0].auto_created
+                except ValueError:
+                    pass
```

Comment:
I think this logic would mark `Children.parent_ptr.auto_created = True` when it should be `False`

```python
class Parent(models.Model):
    pass

class Children(Parent):
    parent_ptr = models.OneToOne(Parent, parent_link=True)
```

Reply:
Should the check logic be adapted to deal with `parent_link` primary key differently instead?

Reply:
Yes, I think you are right. 
I tried to check `field.remote_field.parent_link`, but it was `True` for both your mentioned test and the test that I added to the patch.

I don't know how we can distinguish between these two cases.

Reply:
I think we should ignore inherited PKs and check them only in parents, see #13925.